lib/seahorn/BvOpSem2.cc

```diff
@@ -914,12 +940,19 @@ class OpSemMemManager {
     for (unsigned i = 0; i < len; i += sem_word_sz) {
       Expr offset = bv::bvnum(i, ptrSzInBits(), m_efac);
       Expr dIdx = i == 0 ? dPtr : mk<BADD>(dPtr, offset);
-      unsigned word = 0;
+      // copy bytes from buffer to word - word must accommodate largest supported word size
+      unsigned long word = 0;  // 8 bytes because assumed largest supported sem_word_sz = 8
       for (int byte = sem_word_sz - 1; byte >= 0; byte--) {
-        word = word << 8;
+        word = word << 8L;
         if (i + byte < len)
           word += sPtr[i + byte];
```

Comment:
is this addition representing a logical OR?
This loop should be extracted into a function. 
Best way to implement copying bytes into words is to use `memcpy`